### DISCUSSION TITLE
Nature Renderer Vegetation Ambient Lighting

### DISCUSSION THREAD

**Discussion by NicholasBrown-01** [visualdesigncafe/nature-renderer#6] [Help] [answered]
I cannot seem to find a setting that can increase the distance in which vegetation that has been converted will render light/ambient light that affects it. I have included a screenshot of being a certain distance from an object, followed by the next screen shot of moving closer a few paces. If I deactivate the Nature Renderer Script attached to the terrain, the problem goes away and all vegetation meshes/textures render ambient lighting correctly. Any help is appreciated thank you!
![1](https://github.com/visualdesigncafe/nature-renderer/assets/119603466/3cb31db7-5a85-4fef-b54c-b81e66c2b0a8)
![2](https://github.com/visualdesigncafe/nature-renderer/assets/119603466/437e7cd2-2ed3-4f1b-af97-223423a55581)
![3](https://github.com/visualdesigncafe/nature-renderer/assets/119603466/81f000b0-4e97-4dd6-83c9-2498ba16d5e0)

**Comment by krazyxiii** [comment]
Which tree asset are you using? I think I had that issue before but it had to do with the material. I can have a look if i have the same tree asset package.

**Reply by NicholasBrown-01** [reply]
Hello, so I played around with this more. At first I thought it was an issue with when NatureRenderer clones the tree and applies the NaturerRenderer shader to it, but if I place the direct tree prefab from my assett folder in to the scene as an individual gameobject with the NatureRenderer converted shader it works. Thus the ambient light issue only occurs on trees/grass etc placed via the terrain tools such as paint trees or paint details for grass.  I have tried Triforge tree assets and The vegetation engine trees the problem occurs with both and applies to all meshes/objects placed with terrain tools. I attached a picture with the prefab working at all times on the left while the terrain placed tree does not receive lighting until closer. (I am showing the exact point/distance where the unlit tree switches from unlit to lit...ignore the light post not being lit as this is an occlusion issue as i just happened to place it in the way of an object....it remains light at the same distance as the indvidually placed tree prefab from other angles)
![Tree1](https://github.com/visualdesigncafe/nature-renderer/assets/119603466/fe5533a4-ee56-4839-8be4-bdc56aabe558)
![Tree2](https://github.com/visualdesigncafe/nature-renderer/assets/119603466/4c5860e0-724a-4f32-977d-9c1c1619c1be)
![Inspector](https://github.com/visualdesigncafe/nature-renderer/assets/119603466/1d3e7430-d997-4ccc-aa4e-162f6065533b)

**Comment by krazyxiii** [comment]
I will look into it some more, hopefully tomorrow I have more time. I don't think that I have that issue with the light, maybe with shadows I thought I saw something earlier. Anyway, I don't even know if the Developer would look at these issues or if we need to log a "Bug" in another forum. Have you reached out to him directly? I did before, had sent an email sometime ago but the auto reply directed me to this board. Anyway like I said I will try to reproduce your issue and see if maybe one of us gets lucky and figures out how to fix these issues. Ultimately I may have to go with an alternate asset that does the same as this. I would like to use my time more effectively if I can't figure this out quickly and don't see a reply from the dev. I'll keep you posted, and good luck!

**Comment by maxplooi** [comment, marked as answer]
I replied on your reported issue (#7) regarding this. It is most likely caused by the _Per Object limit_ setting for _Additional Lights_ in the URP settings. Changing the renderer to **Forward+** is the best solution since it removes the limit on the number of lights. Let me know if this does not solve the issue and I'll look into it further.
